### ISSUE TITLE
[Bug]:  Fluent UI react v9 Combox - Talkback is not reading the Combox options and it is also not opening the listbox when user starts typing in it

### ISSUE THREAD

**Issue by v-prigunasek** [microsoft/fluentui#27897]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
Screen reader: Talkback and Voiceover
```

### Are you reporting Accessibility issue?

None

### Reproduction

React Components / v9 - https://codesandbox.io/

## Actual Behavior
We are having some accessibility issues with Fluent UI react v9 Combo Box control.
1. In Talkback the list box is not opening when the user starts typing in it. It gets opened only after when we double tap on the dropdown icon.
2. Once the list box is opened, focus is moving into the combo box options with swipe gesture, but it is not reading the values, instead it is reading the data of the background controls.
![MicrosoftTeams-image (7)](https://github.com/microsoft/fluentui/assets/98514691/a77276ca-ba83-4b69-ad0b-0ba945a01d96)
3. In both Voiceover and Talkback - the position for the combo box options is not announced.

## Expected Behavior
1. In Talkback, the list box should be opened when the user starts typing in it.
2. In Talkback, focus should move into the combo box options and read out the values.
3. In both Voiceover and Talkback - the position for the combo box options should be announced.

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by Hotell** [comment]
can you please provide demo on codesandbox to help us out troubleshooting quicker ? if there wont be a demo provided we will close this issues. ty
